packages/caver-klay/caver-klay-accounts/src/index.js

```diff
@@ -191,50 +191,85 @@ Accounts.prototype.getLegacyAccount = function getLegacyAccount(key) {
     var _this = this,
-        error = false,
         isLegacy = false,
-        parsed,
-        result
+        result,
+        callback
 
-    callback = callback || function () {}
+    let handleError = (e) => {
+      e = e instanceof Error? e : new Error(e)
+      if (callback) callback(e)
+      return Promise.reject(e)
+    }
 
-    if (!tx) {
-      error = new Error('No transaction object given!')
-      callback(error)
-      return Promise.reject(error)
+    if (arguments.length === 0 || arguments.length > 3) {
+      return handleError('Inavlid parameter: The number of parameter is invalid.')
     }
+    
+    // privateKey and callback are optional parameter
+    // "arguments.length === 2" means that user sent parameter privateKey or callback
+    let tx = arguments[0], privateKey
 
-    try {
-      parsed = utils.parsePrivateKey(privateKey)
-      privateKey = parsed.privateKey
-    } catch(e) {
-      callback(e)
-      return Promise.reject(e)
+    if (!tx || !_.isObject(tx)) {
+      return handleError('Invalid paramter: The transaction must be defined as an object')
     }
-    if (!utils.isValidPrivateKey(privateKey)) {
-      error = new Error('Invalid private key')
-      callback(error)
-      return Promise.reject(error)
+
+    if (arguments.length === 2) {
+      if (_.isFunction(arguments[1])) {
+        callback = arguments[1]
+      } else {
+        privateKey = arguments[1]
+      }
+    } else if (arguments.length === 3) {
+      if (typeof arguments[1] !== 'string' && !_.isArray(arguments[1])){
+        return handleError('Invalid paramter: The parameter for the private key is invalid')
+      }
+      privateKey = arguments[1]
+      callback = arguments[2]
     }
 
-    privateKey = utils.addHexPrefix(privateKey)
+    // For handling when callback is undefined.
+    callback = callback || function () {}
 
-    function signed(tx) {
+    // Validate tx object
+    if (!tx.senderRawTransaction) {
+      const error = helpers.validateFunction.validateParams(tx)
+      if (error) return handleError(error)
+    }
 
-      if (!tx.senderRawTransaction) {
-        error = helpers.validateFunction.validateParams(tx)
-        
-        // Attempting to sign with decoupled account into a legacy type transaction throw an error.
-        isLegacy = tx.type === undefined || tx.type === 'LEGACY' ? true : false
-        if (!error && isLegacy && _this.isDecoupled(privateKey, tx.from)) error = new Error('A legacy transaction must be with a legacy account key')
+    // When privateKey is undefined, find Account from Wallet.
+    if (privateKey === undefined) {
+      try {
+        const account = this.wallet.getAccount(tx.from || tx.feePayer)
+        privateKey = account.privateKey
+      } catch(e) {
+        return handleError(e)
       }
-      if (error) {
-        callback(error);
-        return Promise.reject(error);
+    }
+
+    let privateKeys = _.isArray(privateKey) ? privateKey : [privateKey]
```

Comment:
_.isArray(privateKey)?

Is this right? I think privateKey is always an array like you checked before at line 225:
```
_.isArray(privateKey)
```

Reply:
In line 225, if privateKey is not string or array, return error.
So, privateKey can be a string type. :)

Reply:
Also, privateKey can be an array, too, based on the condition:

```js
 if (typeof arguments[1] !== 'string' && !_.isArray(arguments[1])){
```

Reply:
Yes sure, privateKey can be an array too ! 
So this line is for converting privatekey to Array.
Check privateKey is an Array or not, and if not, make privateKey to Array type.

Reply:
Got it.